### ISSUE TITLE
Follow XDG Base Directory Specification

### ISSUE THREAD

**Issue by maujim** [ItayGarin/ktrl#34]
Can you change the defaults to follow the XDG Base Directory Specification? 

Full spec is [here](https://specifications.freedesktop.org/basedir-spec/basedir-spec-latest.html) but the main changes would just be to change the assumption that configs and assets live in `/opt/ktrl` to the specified dirs.

It would only involve the following changes:	
- assume config is found in `$XDG_CONFIG_HOME/ktrl/cfg.ron` if the variable is set and `$HOME/.config/ktrl/cfg.ron` otherwise
- assume assets found at `$XDG_DATA_HOME/ktrl/assets` if the variable is set and `$HOME/.local/share/ktrl/assets` otherwise
- default to logging in `$XDG_CACHE_HOME/ktrl/log.txt` if the variable is set and `$HOME/.cache/ktrl/log.txt` otherwise

**Comment by ItayGarin** [comment]
Hey Mukund! Thanks for the suggestion :)
Wasn't aware of the XDG spec. Glad you brought it to my attention!

I might be missing something, but since ktrl requires high privileges, 
it isn't intended to be run using a "normal" user (having a home directory).
It should ideally run as a systemd daemon (using the privileged `ktrl` user).

Do you think it still makes sense to follow the XDG spec in this case?
Of course, one could always override these defaults using CLI arguments.

**Comment by maujim** [comment]
I think it would still make sense to follow the spec. It's possible the run the systemd service from the current user, using the `--user` flag. I can test this out and get back to you to see if that causes any problems

**Comment by maujim** [comment]
I think it'd be hard to follow the spec since ktrl needs to run as the `ktrl` user. From my understanding, I don't think `systemd` lets you run a service with the `--user` flag if it runs as a different user. 

I think it's still possible to at least follow the default config file location. Further, would it be possible to include the target keyboard via `cfg.ron` instead of the `-d` flag?

I would like to help out more but I'm also getting the same issue as https://github.com/ItayGarin/ktrl/issues/32
